plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts

```diff
@@ -0,0 +1,373 @@
+  ): RegisterFunction<T> {
+    const instrumentation = this;
+    this._logger.debug('Patching Hapi.Server register function');
+    return function register(
+      this: Hapi.Server,
+      pluginInput: HapiPluginInput<T>,
+      options?: Hapi.ServerRegisterOptions
+    ) {
+      if (Array.isArray(pluginInput)) {
+        for (let i = 0; i < pluginInput.length; i++) {
+          instrumentation._wrapRegisterHandler(pluginInput[i].plugin);
+        }
+      } else {
+        instrumentation._wrapRegisterHandler(pluginInput.plugin);
+      }
+      return original.call(this, pluginInput, options);
+    };
+  }
+
+  /**
+   * Patches the Server.ext function which adds extension methods to the specified
+   * point along the request lifecycle. This function accepts the full range of
+   * accepted input into the standard Hapi `server.ext` function. For each extension,
+   * it adds instrumentation to the handler via a call to the @function _wrapExtMethods
+   * function.
+   * @param original - the original ext function which adds the extension method to the server
+   * @param {string} [pluginName] - if present, represents the name of the plugin responsible
+   * for adding this server extension. Else, signifies that the extension was added directly
+   */
+  private _getServerExtPatch(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    original: (...args: any) => void,
+    pluginName?: string
+  ) {
+    const instrumentation = this;
+    this._logger.debug('Patching Hapi.Server ext function');
+    return function ext(
+      this: Hapi.Server,
+      ...args: Parameters<typeof original>
+    ) {
+      if (Array.isArray(args[0])) {
+        const eventsList:
+          | Hapi.ServerExtEventsObject[]
+          | Hapi.ServerExtEventsRequestObject[] = args[0];
+        for (let i = 0; i < eventsList.length; i++) {
+          const eventObj:
+            | Hapi.ServerExtEventsObject
+            | Hapi.ServerExtEventsRequestObject = eventsList[i];
+          if (isLifecycleExtType(eventObj.type)) {
+            const lifecycleEventObj = eventObj as Hapi.ServerExtEventsRequestObject;
+            const handler = instrumentation._wrapExtMethods(
+              lifecycleEventObj.method,
+              eventObj.type,
+              pluginName
+            );
+            lifecycleEventObj.method = handler;
+            eventsList[i] = lifecycleEventObj;
+          }
+        }
+      } else if (args.length == 1) {
+        const eventObj:
+          | Hapi.ServerExtEventsObject
+          | Hapi.ServerExtEventsRequestObject = args[0];
+        if (isLifecycleExtType(eventObj.type)) {
+          const lifecycleEventObj = eventObj as Hapi.ServerExtEventsRequestObject;
+          const handler = instrumentation._wrapExtMethods(
+            lifecycleEventObj.method,
+            eventObj.type,
+            pluginName
+          );
+          lifecycleEventObj.method = handler;
+          args[0] = lifecycleEventObj;
```

Comment:
This would change the args parameter by reference. Is this expected? If it only needs to be modified to make the call below I would consider local copy to protect the caller from unexpected modifications.

Reply:
Following up on our conversation earlier, I have checked over all of these cases of modifying by reference and determined that they all are as expected! I did make some minor changes for readability/general code updates